### PR TITLE
Proxy fix

### LINKED ISSUE #187: Kismatic Inspector should use the internal IP
Currently the inspector checks if ports are accessible on the public interface, this is not needed as all cluster communication is done via the internal interface (TODO verify this statement is correct)

```
➜  out git:(master) ./kismatic install validate

Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [OK]
1/3....  Configure Cluster Prerequisites                                        [OK]
2/3....  Update Hosts File                                                      [OK]
3/3....  Run Cluster Pre-Flight Checks
- Running task: run pre-flight checks using Kismatic Inspector
  node1                                                                         [ERROR]
---- STDERR ----
Error: error running inspector against remote node: error posting request to server: Post http://130.211.120.125:8888/execute: dial tcp 130.211.120.125:8888: getsockopt: operation timed out
---------------
- Running task: run pre-flight checks using Kismatic Inspector
  node2                                                                         [ERROR]
---- STDERR ----
Error: error running inspector against remote node: error posting request to server: Post http://104.197.142.192:8888/execute: dial tcp 104.197.142.192:8888: getsockopt: operation timed out
---------------
➜  out git:(master) ./kismatic install validate

Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [ERROR]
- Worker nodes: Node #1: Invalid IP provided
validation error prevents installation from proceeding
➜  out git:(master) ./kismatic install validate

Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [OK]
1/3....  Configure Cluster Prerequisites                                        [OK]
2/3....  Update Hosts File                                                      [OK]
3/3....  Run Cluster Pre-Flight Checks
- Running task: run pre-flight checks using Kismatic Inspector
  node1                                                                         [ERROR]
---- STDERR ----
Error: error running inspector against remote node: error posting request to server: Post http://104.197.142.192:8888/execute: dial tcp 104.197.142.192:8888: getsockopt: operation timed out
---------------
- Running task: run pre-flight checks using Kismatic Inspector
  node2                                                                         [ERROR]
---- STDERR ----
Error: error running inspector against remote node: error posting request to server: Post http://104.197.227.252:8888/execute: dial tcp 104.197.227.252:8888: i/o timeout
---------------
➜  out git:(master) ./kismatic install validate
# After opening port 8888 on the public interface
Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [OK]
1/3....  Configure Cluster Prerequisites                                        [OK]
2/3....  Update Hosts File                                                      [OK]
3/3....  Run Cluster Pre-Flight Checks
=> The following checks failed on "node1":
   - Port Accessible: 2379: Port 2379 on host "104.197.142.192" is unreachable. Error was: dial tcp 104.197.142.192:2379: i/o timeout
   - Port Accessible: 6666: Port 6666 on host "104.197.142.192" is unreachable. Error was: dial tcp 104.197.142.192:6666: i/o timeout
   - Port Accessible: 2380: Port 2380 on host "104.197.142.192" is unreachable. Error was: dial tcp 104.197.142.192:2380: i/o timeout
   - Port Accessible: 6660: Port 6660 on host "104.197.142.192" is unreachable. Error was: dial tcp 104.197.142.192:6660: i/o timeout
   - Port Accessible: 6443: Port 6443 on host "104.197.142.192" is unreachable. Error was: dial tcp 104.197.142.192:6443: i/o timeout
   - Port Accessible: 8443: Port 8443 on host "104.197.142.192" is unreachable. Error was: dial tcp 104.197.142.192:8443: i/o timeout

error running preflight: error running playbook: error running ansible: exit status 2
➜  out git:(master) ./kismatic install validate
# After opening all ports on the public interface
Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [OK]
1/3....  Configure Cluster Prerequisites                                        [OK]
2/3....  Update Hosts File                                                      [OK]
3/3....  Run Cluster Pre-Flight Checks                                          [OK]
```

### LINKED ISSUE #744: HTTP proxy, etcd play
Currently the proxy variables are defined at higher level plays `_etcd.yaml`, this is causing an issue when running etcd cluster health if the etcd IPs are not specified in `no_proxy`.

The proxy variables should only be set on the tasks that need it.

### LINKED ISSUE #745: HTTP proxy, preflight
KET does not set the `no_proxy` setting when running preflights. In scenarios where the install nodes are also behind a proxy, that proxy is being used to connect to the other nodes in the cluster where in most cases it should not be.

This requires some though how to solve this.
1) Use the same configuration as the rest of the cluster
2) Use different configuration fields for the `preflight` 
3) Run the preflight client from within the cluster. This would also fix https://github.com/apprenda/kismatic/issues/187

### LINKED ISSUE #746: HTTP proxy, add worker
When using the proxy settings, it is most likely required to set the `no_proxy` flag equal to a list of all IPs and Hostnames of the nodes.

However, when using the `add-worker` command, this will break unless the user modified the plan file prior to running the install.

KET should verify that the new node is in the `no_proxy` list and append to it, or at least fail the validation.

### LINKED ISSUE #762: Proxy e2e test is failing with helm issue
The proxy e2e test is fails with the following error:

```
- Running task: run helm init
  ip-10-0-3-75                                                                  [OK]
- Running task: add kismatic helm repo
  ip-10-0-3-75                                                                  [ERROR]
---- STDERR ----
Error: Looks like "https://apprenda.github.io/kismatic-charts/" is not a valid chart repository or cannot be reached: Get https://apprenda.github.io/kismatic-charts/index.yaml: proxyconnect tcp: dial tcp 10.0.3.250:3128: getsockopt: no route to host
---------------
Initialize Helm and Start Tiller Finished
error installing: error running playbook: error running ansible: exit status 2
```

It is interesting that the helm init step did not fail, as it also should have attempted to go through the proxy.

Failure from:
https://circleci.com/gh/apprenda/kismatic/670?utm_campaign=chatroom-integration&utm_medium=referral&utm_source=slack

### PR DESCRIPTION
**Pull request by dkoshkin**
Fixes #744, fixes #745, fixes #746, fixes #187, fixes #762 

- Move `http_proxy` settings under the tasks that use them
- Run preflight checks from the cluster, not the install node
- Verify connectivity to helm repos during preflight
- Fix disconnected preflight check and run `--preflight` during tests

### CODE REVIEW

**Comment by alexbrand** [comment]
I am somewhat wary about adding all the node hostnames and IPs to NOPROXY. I might have suggested it but I am not sure now. Are we certain that there is no scenario in which going through a proxy for the nodes is desired?

**Comment by dkoshkin** [comment]
I have a similar feeling, I also dont like that we are making a change with no way to go back to the previous behavior. 

I'm thinking we need another option, or some indicator in the `cluster.networking.no_proxy` field.

**Comment by alexbrand** [comment]
Given that the preflight checks are run from within the cluster, we no longer need to add all nodes to the NO_PROXY list. I believe that was originally needed because the installer node was trying to reach nodes through the proxy when trying to communicate with the inspector running on each node.

**Comment by dkoshkin** [comment]
Due to the preflight we still need to have the list of IPs and hosts set in `NO_PROXY`. However, we might need to provide a method of not setting it to maintain support of previous functionality and any installs that do use a proxy for node to node communication.

**Comment by alexbrand** [comment]
@dkoshkin and I discussed this offline. We arrived at the conclusion that including all nodes in the `NO_PROXY` environment variable is still required to prevent preflight checks going through the proxy. 

Furthermore, we agreed that the proxy configuration that is provided to KET is to be used by KET to:
1. Set the proxy on any command that performs HTTP/HTTPS requests on resources outside of the cluster nodes, such as downloading packages, images, etc.
2. Configure the proxy on any component that performs HTTP/HTTPS requests on resources outside of the cluster nodes, such as the docker engine when pulling images from the internet.

File: pkg/install/execute.go [hunk 1]
```diff
@@ -807,6 +806,19 @@ func (ae *ansibleExecutor) buildClusterCatalog(p *Plan) (*ansible.ClusterCatalog
 	return &cc, nil
 }
 
+func nodeAddresses(nodes []Node) string {
```
Comment (alexbrand) [inline pkg/install/execute.go]: Doesn't seem to be used.

File: ansible/roles/preflight/tasks/main.yaml [hunk 2]
```diff
@@ -73,10 +73,28 @@
...
 
+    # Run from the install node, 
+    # Check if the helm repos can be reached
+  - name: verify install node can reach official helm chart repo
+    uri:
+      url: https://kubernetes-charts.storage.googleapis.com
+    delegate_to: 127.0.0.1
+    become: no
+    run_once: true
+    when: disconnected_installation|bool != true
```
Comment (alexbrand) [inline ansible/roles/preflight/tasks/main.yaml]: seems like this should not run when the helm add-on is disabled?

File: ansible/roles/packages-etcd/tasks/main.yaml [hunk 3]
```diff
@@ -11,6 +11,10 @@
     retries: 3
     delay: 3
     when: ansible_os_family == 'RedHat'
+    environment:
```
Comment (alexbrand) [inline ansible/roles/packages-etcd/tasks/main.yaml]: Is it not possible to set the proxy options at the role level?

File: ansible/roles/preflight/tasks/main.yaml (hunk 2 above)
Comment (dkoshkin) [reply ansible/roles/preflight/tasks/main.yaml]: Ahh good call, forgot about that option.

File: ansible/roles/packages-etcd/tasks/main.yaml (hunk 3 above)
Comment (dkoshkin) [reply ansible/roles/packages-etcd/tasks/main.yaml]: No only for the whole play or task, no role level.

**Comment by alexbrand** [comment]
LGTM

**Pull request merged**